Kernel/System/PostMaster/FollowUp.pm

```diff
@@ -324,22 +325,69 @@ sub Run {
         );
     }
 
-    # set ticket service
+    # Ticket service handling.
+
     if ( $GetParam{'X-OTRS-FollowUp-Service'} ) {
 
-        $TicketObject->TicketServiceSet(
-            Service  => $GetParam{'X-OTRS-FollowUp-Service'},
-            TicketID => $Param{TicketID},
-            UserID   => $Param{InmailUserID},
-        );
+        my $ServiceObject = $Kernel::OM->Get('Kernel::System::Service');
 
-        $Self->{CommunicationLogObject}->ObjectLog(
-            ObjectLogType => 'Message',
-            Priority      => 'Debug',
-            Key           => 'Kernel::System::PostMaster::FollowUp',
-            Value =>
-                "Services update via 'X-OTRS-FollowUp-Service'! Service: $GetParam{'X-OTRS-FollowUp-Service'}.",
+        # Check if service exists
+        my $ServiceID = $ServiceObject->ServiceLookup(
+            Name => $GetParam{'X-OTRS-FollowUp-Service'},
         );
+
+        if ($ServiceID) {
+
+            # If service with given name exists, don't set it if not active.
+
+            # Get all active services.
+            my %ServiceList = $ServiceObject->ServiceList(
+                Valid        => 1,
+                KeepChildren => $ConfigObject->Get('Ticket::Service::KeepChildren') // 0,
+                UserID       => $Param{InmailUserID},
+            );
+
+            if ( !$ServiceList{$ServiceID} ) {
+                $Self->{CommunicationLogObject}->ObjectLog(
+                    ObjectLogType => 'Message',
+                    Priority      => 'Debug',
+                    Key           => 'Kernel::System::PostMaster::FollowUp',
+                    Value =>
+                        "Ticket service won't be updated to '$GetParam{'X-OTRS-FollowUp-Service'}' (service invalid or is a child of invalid service).",
+                );
+                $GetParam{'X-OTRS-FollowUp-Service'} = '';
+            }
+        }
```

Comment:
`ServiceLookup()` is too much at this point, as you are still checking for validation.
Please use ServiceList() and reverse the hash:

`name => id`

Then you can check the name directly:

`$ServiceList{$GetParam{'X-OTRS-FollowUp-Service'}}`
Then please adjust the logs accordingly..

Reply:
done